spec/interactors/pay_cart_spec.rb

```diff
@@ -153,5 +159,39 @@
           to_not change(user, :balance)
       end
     end
+
+    context 'when things come in fast', :transactional do
+
+      it 'shouldn\'t do things twice' do
+        expect {
+          threaded(3) do
+            PayCart.new(
+              cart_id: 1,
+              cart_version: 3,
+              requesting_application: application
+            ).call rescue Interactor::Failure
+          end
+        }.to change(user, :balance).from(1000).to(700)
+      end
+    end
+  end
+
+  def threaded(count, &block)
+    threads = []
+    count.times do |i|
+      threads << Thread.new(i) do
+        ActiveRecord::Base.connection_pool.with_connection do
+          begin
+            yield
+          rescue Exception => e
```

Comment:
Avoid rescuing the `Exception` class. Perhaps you meant to rescue `StandardError`?